src/js/control-bar/text-track-controls/off-text-track-menu-item.js

```diff
@@ -55,18 +55,26 @@ class OffTextTrackMenuItem extends TextTrackMenuItem {
    */
   handleTracksChange(event) {
     const tracks = this.player().textTracks();
-    let selected = true;
+    let shouldBeSelected = true;
 
     for (let i = 0, l = tracks.length; i < l; i++) {
       const track = tracks[i];
 
       if ((this.options_.kinds.indexOf(track.kind) > -1) && track.mode === 'showing') {
-        selected = false;
+        shouldBeSelected = false;
         break;
       }
     }
 
-    this.selected(selected);
+    if (shouldBeSelected) {
```

Comment:
If I understand correctly, that would create scenarios where `shouldBeSelected` is true, `isSelected_` is true, yet we would deselect the menu item. In that case we should just do nothing.

Reply:
Ah, right, makes sense. What about moving the check into the `this.selected` method?

Reply:
Yeah I considered it. It was kind of a toss up, but in the end I decided I found it clearer to conditionally call `.selected()` rather than always call it and conditionally have it do nothing, if that makes sense. I don't have a particularly strong preference, so if you find it clearer to have the logic in the selected method itself I'm happy to move it.

Reply:
Nah, don't really have much preference one way or the other. @videojs/core-committers do you have any thoughts?

Reply:
Since we're making this change, it seems unnecessary to allow redundant `selected(false)` calls as well. How about `shouldBeSelected && shouldBeSelected !== this.isSelected_)` for the first if and `!shouldBeSelected && shouldBeSelected !== this.isSelected_` for the second?

Reply:
That sounds good to me

Reply:
Or even this: 

```
if (shouldBeSelected !== this.isSelected_) {
  this.selected(shouldBeSelected);
}
```